conversion.qmd

```diff
@@ -2,3 +2,152 @@
 title: "Conversion to and from R data"
 ---
 
+One of the key goals with extendr, is to provide a framework that allows you to write Rust functions, that interact with R, without having to know the intricacies within R internals, or even R's C-facilities. However, this is unavoidable if one wishes to understand why the extendr-api is the way it is.
+
+Thus, for introducing extendr, we shall mention facts about R internals, but these are not necessary to keep in mind going forward.
+
+```{r, echo=FALSE}
+# to enable extendr/extendrsrc chunks
+library(rextendr)
+# names(knitr::knit_engines$get())
+```
+
+A fundamental data-type in R is the 32-bit integer, `int` in C, and `i32` in Rust. Passing that type around is essential, and straight forward:
+
+```{extendrsrc}
+#[extendr(use_try_from = true)]
+fn ultimate_answer() -> i32 {
+    return 42_i32;
+}
+```
+
+And now this function is available within your R-session, as the output is `r ultimate_answer()`.
```

Comment:
i like starting with this example, but maybe show output using an R code chunk

````
```{r}
ultimate_answer()
```
````